lib/src/components/NxVulnerabilityIndicator/NxVulnerabilityIndicator.scss

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+ @use '../../scss-shared/nx-dark-mode-helpers';
+ 
+// eyeballed, assumes $nx-font-size surrounding text
+$-vertical-alignment: -0.1em;
+
+.nx-vulnerability-indicator {
+  $-icon-width: 12.25px;
+  $-font-size: var(--nx-font-size-s);
+
+  font-size: $-font-size;
+  // This adds side bearing to the icon
+  // so that it aligns with the fa-circle
+  // in the NxThreatIndicator.
+  margin: 0 calc(
```

Comment:
This still seems sketchy to me.  There's nothing in this formula that inherently ties it to the width of the circle icon (unless there's an assumption that the circle's diameter equals the font size), it just kind of happens to work.  If we really want the icons to align (which I still think is an unnecessary requirement), I'd think the right way would be to use FA's `fixedWidth` mode.

Reply:
> This still seems sketchy to me. There's nothing in this formula that inherently ties it to the width of the circle icon (unless there's an assumption that the circle's diameter equals the font size), it just kind of happens to work. If we really want the icons to align (which I still think is an unnecessary requirement), I'd think the right way would be to use FA's `fixedWidth` mode.

I guess I should've rephrased the comment so it's not directly tied with making it aligned with NxThreatIndicator but visually consistency with this this type/category of component (fa icon next to label as part in this style). The fontsize is a good anchor for this because it's what determines the canonical boundary size (em square size of the icon) not just the diameter of the circle of fa-circle. This was discussed with Kiran but let me reach out with Jason to get their opinion on this.

And depending on their decision I can remove it or change it to `fixedWidth` mode
and update the comment to reflect the intent better.

Reply:
> The fontsize is a good anchor for this because it's what determines the canonical boundary size (em square size of the icon) not just the diameter of the circle of fa-circle

Well, except that as demonstrated by the square here, FA icons are not at all guaranteed to be 1em in width.

Reply:
> > The fontsize is a good anchor for this because it's what determines the canonical boundary size (em square size of the icon) not just the diameter of the circle of fa-circle
> 
> Well, except that as demonstrated by the square here, FA icons are not at all guaranteed to be 1em in width.

Yes, but for this specific type of component, it was with the intent of icons within that 1em
for example the square or circle which are simplified and pure shapes.

The FA icons that extend beyond 1em width to me fall into a different category and usage.

Reply:
Sorry, I miss-read and argued the wrong thing here.

Here we're using that 1em as an "anchor" because the spacing around the icon also contributes to how it is perceived. The overall intent is that we want "indicator type" components essentially this and the threat component to be visually weighted the same.

Yes it is not a requirement to make the two "align" but we still need to make them visually weighted equally.
If we remove the side-bearing it will no longer be visually the same because despite the spacing being equal, it will be perceived as such that there is "less" spacing between the square and the circle. causing more tension which unbalance it visually compared to the circle. This is because the circle has more space around its curvature, whereas the square takes up that area.

Reply:
That's quite abstract and makes me uncertain what concrete goals you are trying to solve for.  Are you talking about making the size (area) of the icons match? Or are you talking about making the distance from the nearest edge of the icon to the adjacent text match, or are you talking about making the distance from the center of the icon to the adjacent text match? Different parts of what you've said seem to potentially be alluding to all of these, but I don't think it's possible to have all three match at the same time.  

So I am uncertain what you are trying to do, but I thought, in this thread at least, the question was how to make it so that when we have two lines of content, one starting with a Threat Indicator and one starting with a Vulnerability Indicator, both the centers of the two icons and the beginnings of the two text spans are horizontally aligned, respectively.  This would imply a consistent distance from the center of the icons to their left and right margin box edges.  But we disagree about how that should be implemented, and how effective the current implementation is.  The standard way to accomplish this with with FA's `fixedWidth` setting (plus consistent margins).  What you have currently is an adjustment to the l/r margin of the square icon that is based on an _assumption_ of the width of the circle icon, even though that width is not something that we are controlling.  That is risky, and also complicated. Whether this assumed width is 1em or some other value doesn't really matter, it's still a risky assumption that seems unnecessary given the existence of a more standard mechanism for this.

> but we still need to make them visually weighted equally which is a requirement.

Is it? Why? I thought it was established that we don't actually expect Threat Indicators and Vulnerability Indicators to be used near each other.

Reply:
I also think this isn't the level at which this debate should be had, nor are we the people to have it.  I am still awaiting precise, well documented mockups, where the answers to these questions should be worked by the designers.  Once those exist we can debate how to make the pixels in the implementation match those in the mockup.

Reply:
> The 1em width is a natural anchor because, both shapes are symmetrical, the offset between the top and bottom of the square is the intended weighted offset by the icon designer, and we're just extending that symmetrically on the transverse axis.

OK I think I finally get what you're saying with this part.  You aren't trying to directly make one icon the size of the other, but are more trying to make both icons be `1em` x `1em` in size.  In that case, a simpler implementation of that goal might be to simply set `width: 1em` rather than the complex margin calculation.  And we'd want to do that in Threat Indicator's styles too, rather than relying on an assumption that the circle is already 1em wide.  However, I still contend that the more standard approach would be the `fixedWidth` setting.  It'll depend on what spacing the mockups call for once they are finished.

Reply:
> That's quite abstract and makes me uncertain what concrete goals you are trying to solve for. Are you talking about making the size (area) of the icons match? Or are you talking about making the distance from the nearest edge of the icon to the adjacent text match, or are you talking about making the distance from the center of the icon to the adjacent text match? Different parts of what you've said seem to potentially be alluding to all of these, but I don't think it's possible to have all three match at the same time.
 
Yes the concrete goal is to keep the intended visual weight of the icons and spacing between the icon and tex consistent as the icon designer intended which included the area and the offset between the nearest edge of the icon and adjacent text. On the topic of area, this was already set by the icon designer. They purposefully drew the icon within the 1em square and each shape are relative to that. The area of the square may not match 100% mathematically with the circle, but this was the decision of the icon designer, either artistically or stylistically. We should respect that same as we would with typography so that's why we want to keep the 1em bounding box vertically. The point here is that this also effects the spacing horizontally between other elements and the icon.
Since the icon is symmetrical, I thought the best way to do this was to keep the 1em horizontally as well because the optical offset would be the same.

> So I am uncertain what you are trying to do, but I thought, in this thread at least, the question was how to make it so that when we have two lines of content, one starting with a Threat Indicator and one starting with a Vulnerability Indicator, both the centers of the two icons and the beginnings of the two text spans are horizontally aligned, respectively. This would imply a consistent distance from the center of the icons to their left and right margin box edges. But we disagree about how that should be implemented, and how effective the current implementation is. The standard way to accomplish this with with FA's fixedWidth setting (plus consistent margins). What you have currently is an adjustment to the l/r margin of the square icon that is based on an assumption of the width of the circle icon, even though that width is not something that we are controlling. That is risky, and also complicated. Whether this assumed width is 1em or some other value doesn't really matter, it's still a risky assumption that seems unnecessary given the existence of a more standard mechanism for this.

On the topic of implementation, I do agree to switching to fixedWidth setting is a better approach, I'm just waiting for confirmation from Jason before making that change.

> Is it? Why? I thought it was established that we don't actually expect Threat Indicators and Vulnerability Indicators to be used near each other.

Well, it's part of good UX is strive for consistency, although these two components are not expected to be used near each other, they are of a similar style. Both use the same class of icons, and have a label next to them.
Furthermore, we would be doing the icon designer and design in general a disservice to not optically place this icon with other elements. I guess why would we not, knowing the intent of how the icons are weighted relative to each other we don't want to ignore that.

Reply:
Got confirmation from Jason, updated implementation here: [0ab0aad](https://github.com/sonatype/sonatype-react-shared-components/pull/1199/commits/0ab0aad44d91be976c108e4bf5fb8bae5c5e0a25)

Reply:
Ah sorry Ross, just saw the other two comments, for some reason I didn't see them when I responded.

We had that discussion with Kiran and Jason, but were considering having a wider discussion later down the road on addressing the icon import issues in Figma and alignment issues later after this initiative.
But we did all agree to make both 1em by 1em for this vulnerability indicator though.

Kiran and Jason are currently working together on getting all the stuff into Kaleido up to standard. I know they're quite busy so I think we may get the update towards the end of this week.

In terms of implementation I do agree with your suggestion, I made a commit above but will make further adjustments per your recommendation.